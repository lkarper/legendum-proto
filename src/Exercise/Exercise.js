import React from 'react';
import { Route } from 'react-router-dom';
import Learn from '../Learn/Learn';
import Do from '../Do/Do';

const Exercise = (props) => {
    return (
        <div>
            <Route 
                path="/game/exercise/:chapt/learn"
                component={Learn}
            />
            <Route 
                path="/game/exercise/:chapt/do"
                component={Do}
            />

        </div>
    );
}

export default Exercise;