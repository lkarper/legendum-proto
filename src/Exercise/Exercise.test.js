import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Exercise from './Exercise';

describe('Exercise component', () => {
  it('renders the UI without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Exercise />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const wrapper = mount(
      <BrowserRouter>
        <Exercise />
      </BrowserRouter>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
