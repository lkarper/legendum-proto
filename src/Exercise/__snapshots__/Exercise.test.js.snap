// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercise component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Exercise>
      <Route
        component={[Function]}
        path="/game/exercises/:chapt/learn"
      />
      <Route
        component={[Function]}
        path="/game/exercises/:chapt/do"
      />
    </Exercise>
  </Router>
</BrowserRouter>
`;
